MontyTRC89/TombEngine
# Fast reload

## todo

- [x] I have added a changelog entry to CHANGELOG.md file on the branch/fork (if it is an internal change then it is not needed) 
- [x] Pull request meets the Coding Conventions standards: https://github.com/MontyTRC89/TombEngine/blob/master/CONTRIBUTING.md#coding-conventions

## Links to issue(s) this pull request concerns (if applicable)

N/A

## Description of pull request 

NOTE: Should be used together with [complementary Tomb Editor PR](https://github.com/MontyTRC89/Tomb-Editor/pull/929).

Heavily optimizes level loading times, if same level is loaded (e.g. from the savegame). It is done by changing file format to keep heavy audiovisual data (textures and sounds), as well as static geometry data (meshes, animations and floordata) separately, and bypassing these types of data, if same level is loaded.

Additionally, if same level is reloaded, loading screen is bypassed to allow even faster reloading for more smooth experience.

To test:

- Level saving and reloading
- Objects integrity after reloading the level
- Sound and renderer (make sure that after reloading, there is no visual garbage in level and no garbage sounds)
- Scripting (make sure scripts and their state is reloaded correctly)
